src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -309,41 +303,52 @@ public void setPeriod(String period) {
 
     @Override
     public void visitJavadocToken(DetailNode ast) {
-        if (containsSummaryTag(ast)) {
-            validateSummaryTag(ast);
-        }
-        else if (!startsWithInheritDoc(ast)) {
-            final String summaryDoc = getSummarySentence(ast);
-            if (summaryDoc.isEmpty()) {
-                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
-            }
-            else if (!period.isEmpty()) {
-                final String firstSentence = getFirstSentence(ast);
-                final int endOfSentence = firstSentence.lastIndexOf(period);
-                if (!summaryDoc.contains(period)) {
-                    log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+        final DetailNode inlineSummaryTag = getInlineSummaryTag(ast);
+        if (inlineSummaryTag == null) {
+            if (!startsWithInheritDoc(ast)) {
+                final String summaryDoc = getSummarySentence(ast);
+                if (summaryDoc.isEmpty()) {
```

Comment:
This all seems similar to `validateSummaryTag`. Whats the difference and they can't be merged together?

Reply:
They are quite similar but can't be merged-

1. `validateSummaryTag` uses period at end technique while the other one uses contains.
2. They both log different messages in a branch.
3. The last branch differs in logic

We can make this work somehow with boolean flags but I restrain from using boolean flags.